hal/src/argon/wlan_hal.cpp

```diff
@@ -192,12 +192,12 @@ int wlan_fetch_ipconfig(WLanConfig* conf) {
     CHECK_TRUE(iface, SYSTEM_ERROR_INVALID_STATE);
     unsigned flags = 0;
     CHECK(if_get_flags(iface, &flags));
-    CHECK_TRUE((flags & IFF_UP) && (flags & IFF_LOWER_UP), SYSTEM_ERROR_INVALID_STATE);
     // MAC address
     sockaddr_ll hwAddr = {};
     CHECK(if_get_lladdr(iface, &hwAddr));
     CHECK_TRUE((size_t)hwAddr.sll_halen == MAC_ADDRESS_SIZE, SYSTEM_ERROR_UNKNOWN);
     memcpy(conf->nw.uaMacAddr, hwAddr.sll_addr, MAC_ADDRESS_SIZE);
+    CHECK_TRUE((flags & IFF_UP) && (flags & IFF_LOWER_UP), SYSTEM_ERROR_INVALID_STATE);
```

Comment:
This is not enough. You still need [this](https://github.com/particle-iot/firmware/blob/mesh-develop/hal/network/lwip/esp32/esp32ncpnetif.cpp#L205-L219) block of code to be executed before https://github.com/particle-iot/firmware/blob/mesh-develop/hal/network/lwip/esp32/esp32ncpnetif.cpp#L130 (it should still be left in `upImpl()` as well), otherwise the MAC will only be available after the network interface is brought up at least once. As discussed that block of code should be decoupled into a separate method and should be called from both places.

Reply:
Added.